automated_packaging/prepare_new_changelog.pl

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env perl
+use JSON;
+
+#!usr/bin/perl
+$CODE_PATH=$ARGV[0];
+$VERSION=$ARGV[1];
+$NUMBER_OF_LAST_PR=$ARGV[2];
+$GITHUB_USERNAME=$ARGV[3];
+
+# Necessary to write log date
+my @month_names = qw(January February March April May June July August September October November December);
+my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime(time);
+$year += 1900;
+
+# We have to get pages separately due to API limitation
+@issue_pages=(1,2,3,4,5);
+
+# Move to the packaging folder
+chdir($CODE_PATH) or die "$CODE_PATH does not exist";
+$release_major_minor = substr($VERSION, 0, 3);
+
+# Checkout the master branch
+system("git checkout master");
+
+# Now create a new branch based on master
+system("git checkout -B citus-$VERSION-changelog");
+
+# Read the current changelog
+open(CHANGELOG_FILE, "<CHANGELOG.md") || die "Changelog file not found";
+my @changelog_current_lines = <CHANGELOG_FILE>;
+close(CHANGELOG_FILE);
+
+# Update the changelog file of the debian branch
+open(CHANGELOG_FILE, ">CHANGELOG.md") || die "Can't open the Changelog file !";
+print CHANGELOG_FILE "### citus v$VERSION ($month_names[$mon] $mday, $year) ###\n\n";
+
+foreach my $page_number (@issue_pages) {
+	my $issues_text = `curl -u "$GITHUB_USERNAME" 'https://api.github.com/repos/citusdata/citus/issues?state=closed&page=$page_number'`;
+	my @issues = @{decode_json($issues_text)};
+	my $last_number_has_seen=0;
+
+	foreach my $issue ( @issues ) {
+		%issue_hash = %$issue;
+
+		if($issue_hash{'number'} == $NUMBER_OF_LAST_PR) {
+			$last_number_has_seen=1;
+			last;
+		}
+
+		if(exists($issue_hash{'pull_request'}) && exists($issue_hash{'labels'})) {
```

Comment:
Where is this stuff documented? The recent emails (from Dec 6 and Dec 14) don't discuss labels whatsoever.

Is it really better to have this structure rather than have devs think about `CHANGELOG` entries as they write their code? And during previous releases we've hand-ordered the `CHANGELOG`. This tool won't do that (i.e. it won't know how to put higher-impact things up top).